### PR TITLE
cutecom: 0.50.0 -> 0.51.0

### PR DESCRIPTION
**Pull request by oxzi**
###### Motivation for this change
This PR updates `cutecom` to its current release and changes its upstream repository which has moved to GitLab. Furthermore, its prefixing `QT_PLUGIN_PATH` as discussed in #56921 / 89cfbb9cb27317ef688e0d4e3d5681faaa53f87b.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @bennofs

### CODE REVIEW

**Comment by jonringer** [comment]
there's also a wrapQtApp utility
 @worldofpeace I keep forgetting which one is best practice

**Review by bennofs — APPROVED** [review]

File: pkgs/tools/misc/cutecom/default.nix [hunk 1]
```diff
@@ -1,25 +1,33 @@
...
 
   preConfigure = ''
-    substituteInPlace CMakeLists.txt --replace "#find_package(Serialport REQUIRED)" "find_package(Qt5SerialPort REQUIRED)"
+    substituteInPlace CMakeLists.txt \
+      --replace "#find_package(Serialport REQUIRED)" "find_package(Qt5SerialPort REQUIRED)"
+  '';
+
+  buildInputs = [ qtbase qtserialport cmake makeWrapper ];
+
+  postInstall = ''
```
Comment (worldofpeace) [inline pkgs/tools/misc/cutecom/default.nix]: makeWrapper here  is completely unneeded, `mkDerivation` automatically adds `wrapQtAppsHook` which was made for this exact purpose.

`wrapQtApp` as @jonringer has mentioned is only needed when you are in a situation where wrapping manually is needed.

Comment (worldofpeace) [reply pkgs/tools/misc/cutecom/default.nix]: see long thread at https://github.com/NixOS/nixpkgs/issues/65399

File: pkgs/tools/misc/cutecom/default.nix [hunk 2]
```diff
@@ -1,25 +1,33 @@
...
+    sha256 = "1zprabjs4z26hsb64fc3k790aiiqiz9d88j666xrzi4983m1bfv8";
   };
 
   preConfigure = ''
-    substituteInPlace CMakeLists.txt --replace "#find_package(Serialport REQUIRED)" "find_package(Qt5SerialPort REQUIRED)"
+    substituteInPlace CMakeLists.txt \
+      --replace "#find_package(Serialport REQUIRED)" "find_package(Qt5SerialPort REQUIRED)"
+  '';
+
+  buildInputs = [ qtbase qtserialport cmake makeWrapper ];
```
Comment (jonringer) [inline pkgs/tools/misc/cutecom/default.nix]: cmake should be a nativeBuildInputs, everything else should be taken care of by the libsForQt5.callPackage
```suggestion
  nativeBuildInputs = [ cmake ];
```

File: pkgs/tools/misc/cutecom/default.nix (hunk 1 above)
Comment (geistesk) [reply pkgs/tools/misc/cutecom/default.nix]: Thanks for this information.

File: pkgs/tools/misc/cutecom/default.nix (hunk 2 above)
Comment (geistesk) [reply pkgs/tools/misc/cutecom/default.nix]: I'm not quite sure about your suggested diff. If I am also removing `qtserialport` the build will fail.

Please take a look at the updated PR. Thanks a lot and excuse my late reply.

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
gui launches cleanly

```
[3 built, 2 copied (0.1 MiB), 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/71385
1 package were build:
cutecom
```

File: pkgs/tools/misc/cutecom/default.nix (hunk 2 above)
Comment (worldofpeace) [reply pkgs/tools/misc/cutecom/default.nix]: So those dependencies are needed, the callPackage function just ensure they're from the correct scope. But `cmake` should be native :+1:

File: pkgs/tools/misc/cutecom/default.nix [hunk 3]
```diff
@@ -1,25 +1,28 @@
...
 
   preConfigure = ''
-    substituteInPlace CMakeLists.txt --replace "#find_package(Serialport REQUIRED)" "find_package(Qt5SerialPort REQUIRED)"
+    substituteInPlace CMakeLists.txt \
+      --replace "#find_package(Serialport REQUIRED)" "find_package(Qt5SerialPort REQUIRED)"
   '';
-  buildInputs = [qtbase qtserialport cmake];
 
-  meta = {
+  nativeBuildInputs = [ cmake qtserialport ];
```
Comment (worldofpeace) [inline pkgs/tools/misc/cutecom/default.nix]: `qtserialport` is linked against so it should be `buildInputs` like the previous diff. As stated `cmake` should only be native.

Comment (worldofpeace) [reply pkgs/tools/misc/cutecom/default.nix]: that and re-adding `qtbase`. see https://github.com/NixOS/nixpkgs/pull/71385#discussion_r339192970

File: pkgs/tools/misc/cutecom/default.nix (hunk 2 above)
Comment (geistesk) [reply pkgs/tools/misc/cutecom/default.nix]: Thanks, I have updated the PR

File: pkgs/tools/misc/cutecom/default.nix (hunk 3 above)
Comment (geistesk) [reply pkgs/tools/misc/cutecom/default.nix]: Done. Thanks for your review.

**Review by worldofpeace — APPROVED** [review]

**Comment by worldofpeace** [comment]
backported in 4067ae8582362ff142199b0eb4955dc57ac8b416

**Comment by worldofpeace** [comment]
Thanks for fixing this @geistesk :star2:

**Pull request merged**